core/src/env.rs

```diff
@@ -144,6 +144,12 @@ pub trait Ext {
     /// Refund some gas.
     fn refund_gas(&mut self, amount: u64) -> Result<(), Self::Error>;
 
+    /// Reserve some gas for a few blocks.
+    fn reserve_gas(&mut self, amount: u32, blocks: u32) -> Result<ReservationId, Self::Error>;
```

Comment:
```suggestion
    fn reserve_gas(&mut self, amount: u64, duration: u32) -> Result<ReservationId, Self::Error>;
```

We calculate gas with u64 vars